core/blockchain.go

```diff
@@ -993,7 +993,7 @@ func (bc *BlockChain) rewindPathHead(head *types.Header, root common.Hash) (*typ
 // then block number zero is returned, indicating that snapshot recovery is disabled
 // and the whole snapshot should be auto-generated in case of head mismatch.
 func (bc *BlockChain) rewindHead(head *types.Header, root common.Hash) (*types.Header, uint64) {
-	if bc.triedb.Scheme() == rawdb.PathScheme {
+	if bc.triedb.Scheme() == rawdb.PathScheme && !bc.NoTries() {
 		return bc.rewindPathHead(head, root)
```

Comment:
what if PBSS + Fastode, it will run into `rewindHashHead(head, root)`, is it reasonable?

Reply:
It is a little bit tricky, `rewindHashHead(head, root)` could recover fast node based on its disklayer, may refer:
```
(bc *BlockChain) HasState(hash common.Hash)
```